src/Compilers/CSharp/Portable/Symbols/Synthesized/GeneratedNames.Parser.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using System;
+using System.Globalization;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal static partial class GeneratedNames
+    {
+        internal const char DotReplacementInTypeNames = '-';
+        internal const string AnonymousNamePrefix = "<>f__AnonymousType";
+
+        internal static bool TryParseAnonymousTypeTemplateName(string name, out int index)
+        {
+            // No callers require anonymous types from net modules,
+            // so names with module id are ignored.
+            if (name.StartsWith(AnonymousNamePrefix, StringComparison.Ordinal))
+            {
+                if (int.TryParse(name.Substring(AnonymousNamePrefix.Length), NumberStyles.None, CultureInfo.InvariantCulture, out index))
+                {
+                    return true;
+                }
+            }
+
+            index = -1;
+            return false;
+        }
```

Comment:
Is it necessary to move these methods to a separate file? Moving the methods makes it difficult to track changes.